Python/Product/PythonTools/visualstudio_py_debugger.py

```diff
@@ -371,12 +371,12 @@ def is_file_in_zip(filename):
         return False
     elif parent in KNOWN_ZIPS:
         return True
-    elif path.isdir(parent):
+    elif path.isfile(parent):
```

Comment:
I was wrong about this change - we need to keep splitting the path until we definitively have a path or a file (otherwise we may be in a subdirectory within a ZIP file, which will be false for both isdir and isfile):

```python
def is_file_in_zip(filename):
    parent, name = path.split(path.abspath(filename))
    if parent in KNOWN_DIRECTORIES:
        return False
    elif parent in KNOWN_ZIPS:
        return True
    elif path.isfile(parent):
        KNOWN_ZIPS.add(parent)
        return True
    elif path.isdir(parent):
        KNOWN_DIRECTORIES.add(parent)
        return False
    else:
        return is_file_in_zip(parent)
```

With this change everything looks to be fine against Python 2.6 and 3.5.